### PR TITLE
Remove fake `__map_file` implementation

### PR DESCRIPTION
**Pull request by sbc100**
Fixes: #16985

### CODE REVIEW

File: system/lib/standalone/standalone.c [hunk 1]
```diff
@@ -116,6 +111,10 @@ __attribute__((__weak__)) int __syscall_fcntl64(int fd, int cmd, ...) {
   return -ENOSYS;
 }
 
+__attribute__((__weak__)) int __syscall_fstat64(int fd, intptr_t buf) {
+  return -ENOSYS;
+}
```
Comment (kripken) [inline system/lib/standalone/standalone.c:116]: Unrelated change?

File: tools/system_libs.py [hunk 2]
```diff
@@ -940,6 +940,7 @@ def get_files(self):
           'ctime_r.c',
           'timespec_get.c',
           'utime.c',
+          '__map_file.c',
```
Comment (kripken) [inline tools/system_libs.py:943]: What does the musl version actually do, in our case?

File: system/lib/standalone/standalone.c (hunk 1 above)
Comment (sbc100) [reply system/lib/standalone/standalone.c:116]: The musl versions call stat on the file before mmap.. so this is now needed to support the musl version.

File: tools/system_libs.py (hunk 2 above)
Comment (sbc100) [reply tools/system_libs.py:943]: It pretty simply.. it just does open + fstat + mmap + close.. so it should actually work .. assuming mmap works for that file:

https://github.com/emscripten-core/emscripten/blob/15802518cbfc8ea5c62ccf0bdf35500175338f65/system/lib/libc/musl/src/time/__map_file.c#L7-L23

Making it working standlone mode would require a better fake __syscall_fstat64 above.. but I don't think that is a high priority.

**Review by kripken — APPROVED** [review]
Sounds good, thanks for the details.
